### ISSUE TITLE
HTTP idle-timeout (play.server.http.idleTimeout) seems to be ignored

### ISSUE THREAD

**Issue by almothafar** [playframework/playframework#8286]
### Play Version
2.6.12

### API 
Java

### Operating System

Windows 10 Enterprise 64-bit build 1709

### JDK 8

```
java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)


```
### Library Dependencies
nothing special:

build.sbt:

```
libraryDependencies ++= Seq(
  guice,
  javaWs,
  filters,
  "org.apache.commons" % "commons-text" % "1.2",
  "com.typesafe.play" %% "play-json" % "2.6.9",
  "com.typesafe.play" %% "play-ahc-ws-standalone" % "1.1.6",
  "com.typesafe.play" %% "play-ws-standalone-json" % "1.1.6",
  "com.typesafe.play" %% "play-mailer" % "6.0.1",
  "com.typesafe.play" %% "play-mailer-guice" % "6.0.1",
  "com.wildbit.java" % "postmark" % "1.0.6",
  "com.google.auth" % "google-auth-library-credentials" % "0.9.0",
  "com.google.firebase" % "firebase-admin" % "5.9.0",
  "com.auth0" % "java-jwt" % "3.3.0",
  "org.mapstruct" % "mapstruct-jdk8" % "1.2.0.Final",
  "org.mapstruct" % "mapstruct-processor" % "1.2.0.Final",
  "net.logstash.logback" % "logstash-logback-encoder" % "5.0"
)
```

plugins.sbt:

```
// The Play plugin
addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.12")

```

### Expected Behavior

To get my config to wait 180 seconds not 75 seconds.

### Actual Behavior
I'm getting this warnining and seems my values are not set correctly and still getting the defautl one.
`
2018-03-12 11:42:01 CDT: [debug] - akka.actor.RepointableActorRef - Aborting tcp connection to /0:0:0:0:0:0:0:1:61338 because of upstream failure: akka.http.impl.engine.HttpIdleTimeoutException: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
`

### Reproducible Test Case

I have the following config:

```
play.server {
  akka.requestTimeout = 180 seconds

  http {
    # The idle timeout for an open connection after which it will be closed
    # Set to null or "infinite" to disable the timeout, but notice that this
    # is not encouraged since timeout are important mechanisms to protect your
    # servers from malicious attacks or programming mistakes.
    idleTimeout = 180 seconds
  }
}

```

In the controller I make it wait for 2 minutes:

```
try {
    Thread.sleep(120 * 1000L);
    System.out.println("Waiting done");
} catch (InterruptedException e) {
    e.printStackTrace();
}
```

**Comment by almothafar** [comment]
I even tried:

```
play.client.http.idleTimeout = 180 seconds
akka.http.server.idle-timeout = 180 seconds
akka.http.client.idle-timeout = 180 seconds
```

no luck!

**Comment by marcospereira** [comment]
Hi @almothafar, are you seeing this in dev mode or production mode?

**Comment by marcospereira** [comment]
By the way,

> play.client.http.idleTimeout = 180 seconds

This is not something we support.

> akka.http.server.idle-timeout = 180 seconds

If you want to directly configure Akka HTTP properties in dev mode, you need to [use `devSettings`](https://www.playframework.com/documentation/2.6.x/ConfigFile#Using-with-the-run-command).

**Comment by almothafar** [comment]
@marcospereira I think I found this line of code here (if you mean `server`): https://github.com/playframework/playframework/blob/5c68b44f979a409aec19296f3827d764d38fd9f7/documentation/manual/releases/release26/migration26/Migration26.md#akka-http-server-timeouts

If you mean `client` I know but I was trying something.

And `akka.http.server.idle-timeout` is not working, and I have this in dev and prod.

The problem when the process takes more than 75 seconds it is returning with a failed response, but in the thread stays and continue to work.

**Comment by almothafar** [comment]
Is it related to this issue? 

![image](https://user-images.githubusercontent.com/13887564/37337809-148ee4d6-26be-11e8-86de-715f84075169.png)

**Comment by almothafar** [comment]
OK, I got this to work, with setting the variable in command line instead of the config file `-Dplay.server.http.idleTimeout=180s ` what is the problem?

I'm pretty sure that the variable in the config file is set correctly, I make this line `play.server.http.idleTimeout=180s` in all if my config file, and got nothing effected.

**Comment by gmethvin** [comment]
@almothafar I'm assuming you're in dev mode. As @marcospereira mentioned:

> If you want to directly configure Akka HTTP properties in dev mode, you need to [use `devSettings`](https://www.playframework.com/documentation/2.6.x/ConfigFile#Using-with-the-run-command).

If you read that section, you'll see that it mentions that HTTP server settings don't work with the `run` command.

**Comment by almothafar** [comment]
@gmethvin no, it is in production mode as well.

**Comment by qwefgh90** [comment]
@almothafar Hello. Did you find a solution without devSettings or system properties? 
If there is no other solution, Is there other way to test a timeout of websocket automatically?

**Comment by laguiz** [comment]
For me Play 2.7 in **dev** mode it works if I add `PlayKeys.devSettings += "play.server.http.idleTimeout" -> "infinite"` in build.sbt. 

As mentioned by @gmethvin see https://www.playframework.com/documentation/2.7.x/ConfigFile#Extra-devSettings

**Comment by j-chimienti** [comment]
I am unable to set the idleTimeout in production mode using either application.conf or command line vars.

When app is running, I am seeing the variable is set, but is ignored in production mode. Development works fine

**Comment by j-chimienti** [comment]
@laguiz any luck in production?

**Comment by laguiz** [comment]
@j-chimienti What version of Play? https://www.playframework.com/documentation/2.7.x/Migration26#Akka-HTTP-server-timeouts

**Comment by tiagoaleff** [comment]
Is there a difference between _play.server.http.idleTimeout=180**s**_  and _play.server.http.idleTimeout=180 **seconds**_ ?

**Comment by ihostage** [comment]
@tiagoaleff Nop. See [this section](https://github.com/lightbend/config/blob/master/HOCON.md#duration-format) of HOCON docs for details.

**Comment by tiagoaleff** [comment]
Thanks for the answer

**Comment by mkurz** [comment]
Turns out `play.server.http.idleTimeout` was actually not working for websockets when using the netty backend, see #11420, will be fixed with Play 2.8.19

**Comment by ManolisPapaspyropoulosAUEB** [comment]
> For me Play 2.7 in **dev** mode it works if I add `PlayKeys.devSettings += "play.server.http.idleTimeout" -> "infinite"` in build.sbt.
> 
> As mentioned by @gmethvin see https://www.playframework.com/documentation/2.7.x/ConfigFile#Extra-devSettings

it worked for me!
Many thanks!
